scripts/utils/price-utils.js

```diff
@@ -134,9 +152,47 @@ module.exports = function(web3 = web3, artifacts = artifacts) {
     return true
   }
 
+  const checkNoProfitableOffer = async (order, exchange, globalPriceStorage = null) => {
+    const tokenInfo = fetchTokenInfoFromExchange(exchange, [order.buyToken, order.sellToken])
+    const currentMarketPrice = await getDexagPrice(
+      (await tokenInfo[order.buyToken]).symbol,
+      (await tokenInfo[order.sellToken]).symbol,
+      globalPriceStorage
+    )
+
+    // checks whether the order amount is negligible
+    if (Math.floor(await valueInUSD(order, tokenInfo, globalPriceStorage)) < 1) {
+      console.log("its getting here")
+      return true
+    }
+
+    const marketPrice = toErc20Units(currentMarketPrice, precisionDecimals)
+    const orderPrice = toErc20Units(order.priceNumerator, precisionDecimals)
+      .mul(new BN(10).pow(new BN((await tokenInfo[order.sellToken]).decimals)))
+      .div(new BN(10).pow(new BN((await tokenInfo[order.buyToken]).decimals)))
+      .div(order.priceDenominator)
+    console.log(marketPrice)
+    console.log(orderPrice)
+
+    return marketPrice.lt(orderPrice)
+  }
+
+  const valueInUSD = async (order, tokenInfo, globalPriceStorage = null) => {
+    const currentMarketPrice = await getDexagPrice("USDC", (await tokenInfo[order.sellToken]).symbol, globalPriceStorage)
+
+    return fromErc20Units(
+      toErc20Units(currentMarketPrice, precisionDecimals)
+        .mul(order.sellTokenBalance)
+        .div(new BN(10).pow(new BN((await tokenInfo[order.sellToken]).decimals))),
+      precisionDecimals
+    )
+  }
```

Comment:
What do we do if the token is not listed on dex.ag against USDC?

Reply:
Let's assume for the peer review that it is listed...